lib/suites/LinkedDataSignature.js

```diff
@@ -145,6 +145,14 @@ module.exports = class LinkedDataSignature {
       proof.nonce = options.nonce;
     }
 
+    // add fields from proofPurpose
+    if(options.proofPurpose) {
+      const proofPurpose = options.proofPurpose;
+      proof.proofPurpose = proofPurpose.uri;
+      await proofPurpose.addFieldsToProof(
```

Comment:
I think it would be best to make this a conditional on `options.proofPurposeOptions`.  It is my assumption that `addFieldsToProof` would add properties to proof from `proofPurposeOptions` to the proof.  The existing implementations of this function don't do anything.

The existing implementation *does* have a return value which is not being utilized here.  So it's hard to know if it's intended to mutate `proof` or return a new proof with the options added.

Reply:
@cwebber is there a reason to call addFieldsToProof if options `proofPurposeOptions` is not defined?

Reply:
> I think it would be best to make this a conditional on options.proofPurposeOptions. It is my assumption that addFieldsToProof would add properties to proof from proofPurposeOptions to the proof.

I guess that's probably true, but maybe not necessarily.  I'm struggling to think of an example that wouldn't be gathered from proofPurposeOptions; it could be that such an example would be something dynamic from the environment such as something time related, though I'll admit that dynamic elements are uncomfortable to ponder.

However, even the absence of proofPurposeOptions could warrant calling this function, as the ProofPurpose class may raise an exception at the *absence* of an option.

> The existing implementations of this function don't do anything.

I guess I hadn't gotten to it, but the intention was that ocap-ld.js would use it to add the`capability` field to the proof.  @gannan08 take note for ocap-ld.js :)

>  The existing implementation does have a return value which is not being utilized here. So it's hard to know if it's intended to mutate proof or return a new proof with the options added.

Yes, we should make a decision on mutate or return a new replacement.